.github/actions/go-test-setup/action.yml

```diff
@@ -0,0 +1,42 @@
+name: setup-go-ipfs
+description: install and setup go-ipfs as needed for testing
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install go-ipfs
+      shell: bash
+      run: (cd /tmp && go install github.com/ipfs/go-ipfs/cmd/ipfs@v0.9.1)
+    - name: Initialize go-ipfs
+      shell: bash
+      run: (ipfs init)
+    - name: Run go-ipfs
+      shell: bash
+      run: (ipfs daemon &)
+    - name: Install jq
+      shell: pwsh
+      run: |
+        if ( "${{ matrix.os }}" -eq "windows") {
+          choco install jq
+        } elseif ( "${{ matrix.os }}" -eq "macos") {
+          brew update
+          brew install jq
+        } elseif ( "${{ matrix.os }}" -eq "ubuntu") {
+          sudo apt-get update
+          sudo apt-get install jq
+        }

```

Comment:
@marten-seeman unless I'm mistaken the instructions in https://github.com/protocol/.github#additional-setup-steps are incorrect and you cannot use an `if:` in a composite. As a result this seems like what you have to do. If so we should update the readme instructions.

Reply:
I think that's correct, we should update the README.
I don't think you need to install `jq` though, I'm pretty sure it's preinstalled (see for example this workflow using it: https://github.com/protocol/.github/blob/master/.github/workflows/check-3rd-party.yml).

Reply:
You can check the preinstalled software here: https://github.com/actions/virtual-environments#available-environments (see included software column).

And you're right, there's `jq 1.5` on `Ubuntu 18.04.6 LTS` and `jq 1.6` on all other images listed there.

Reply:
This workaround is not needed any more, see https://github.blog/changelog/2021-11-09-github-actions-conditional-execution-of-steps-in-actions.

Reply:
Also, it looks like the bot wiped the custom actions added here (like go-ipfs setup for example). Let me restore it in a separate PR.

Reply:
Yes, unfortunately the bot still does that: https://github.com/protocol/.github/issues/114.

Reply:
There it is: https://github.com/ipfs-shipyard/git-remote-ipld/pull/37